### PR TITLE
[2.x] WFCORE-1374 - backport jdk9 related changes from master

### PR DESCRIPTION
**Pull request by ctomc**
- Update compiler config for JDK9 b118+
- WFCORE-1374 - make code work with jdk9b115 and newer
- upgrade jboss-modules to 1.5.2.Final
- upgrade jboss-marshalling to 1.4.11.Final
- [WFCORE-1490] JDK9 doesn't provide sun.boot.class.path attribute
- [WFCORE-1491] Don't rely on default (JDK) JAXP providers - specify explicit ones
- [WFCORE-1493] Fix proper -Djdk.launcher.addexports.%d JVM property values propagation
- [WFCORE-1431] Define and propagate ${modular.jdk.args} property where necessary
- Properly create writers with charset and cleanup resources

### CODE REVIEW

File: testsuite/shared/src/main/java/org/jboss/as/test/integration/management/util/CustomCLIExecutor.java [hunk 1]
```diff
@@ -195,37 +203,6 @@ public static String execute(File cliConfigFile, String operation, String contro
...
-            try {
-                Thread.sleep(100);
-            } catch (InterruptedException e) {
-            }
-            now = System.currentTimeMillis();
-        } while (now - start < timeout);
-
-        fail("Server did not reload in the imparted time.");
-    }
-
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/management/util/CustomCLIExecutor.java]: Is this related to this work?

Comment (ctomc) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/management/util/CustomCLIExecutor.java]: This is unused method, it was in original commit on master. Just took whole commit...
I can remove it if you want.

Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/integration/management/util/CustomCLIExecutor.java]: Please restore it. On master it was removed as part of the "Improve test runner to allow few more things when running in manual server mode" commit.

File: testsuite/domain/pom.xml [hunk 2]
```diff
@@ -178,7 +177,7 @@
                     <argLine>${jvm.args.ip.client} ${jvm.args.timeouts}</argLine>
 
                     <systemPropertyVariables>
-                        <jboss.options>-ea ${surefire.system.args}</jboss.options>
+                        <jboss.options>${surefire.system.args}</jboss.options>
```
Comment (bstansberry) [inline testsuite/domain/pom.xml]: testsuite/pom.xml overrides the surefire.system.args setting from pom.xml and drops the -ea, so without this the PC and HC processes will not have it.

**Comment by bstansberry** [comment]
Superceded by #1618
